rotary/rules/rotary.smk

```diff
@@ -1,21 +1,31 @@
 # rotary: utilities and workflow for long-read DNA assemblies including circular elements
 # Copyright Jackson M. Tsuji and Lee H. Bergstrand, 2023
 
+import os
 from snakemake.utils import min_version
 
 from pungi.dataset import generate_dataset_from_sample_tsv
 from pungi.utils import symlink_or_compress, is_config_parameter_true
 
 SAMPLE_TSV_PATH = 'samples.tsv'
-SAMPLES = generate_dataset_from_sample_tsv(SAMPLE_TSV_PATH)
 
-if SAMPLES.files_per_sample==3:
-    DATASET_HAS_SHORT_READS=True
+# When running rotary in download mode, there can be situations where samples.tsv is not present in the CWD.
+# We bypass processing samples if no samples.tsv file is found in the CWD.
+if os.path.isfile(os.path.join(os.getcwd(), SAMPLE_TSV_PATH)):
+    SAMPLES = generate_dataset_from_sample_tsv(SAMPLE_TSV_PATH)
+
+    if SAMPLES.files_per_sample==3:
+        DATASET_HAS_SHORT_READS=True
+    else:
+        DATASET_HAS_SHORT_READS=False
+
+    SAMPLE_NAMES = list(SAMPLES.identifiers)
 else:
+    # Some variables are required for evaluating the DAG because they are used in rules.
+    # We mock them if no samples.tsv file is found.
+    SAMPLE_NAMES=['mock_sample_name']
```

Comment:
Nice that you were able to find a workaround for `samples.tsv` if the download command is run.